subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvoker.java

```diff
@@ -215,7 +263,9 @@ private void writeResultsFile(File outputDir, File resultsFile, ImmutableList<Fi
                     }
                 }
                 return builder.build();
-            });
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
```

Comment:
I'd prefer throwing the `IOException` as high as we can before wrapping it. In general it would be best if we could avoid having to wrap it altogether.

Reply:
I don't want to throw checked exceptions from our methods. I think we do the same in other places. What is the advantage of wrapping the exception later?

Reply:
The advantage is that you could end up not having to wrap it at all, as it can be handled as it is. Which is sort of the intention with all this.

Reply:
The `IOException` thrown here is caused by Gradle code, not by user code. So it would cause the build to fail, going through various layers. All the layers in between would be required to declare `throws IOException`/`throws Exception`.

Were would you expect the wrapping to happen?

Reply:
Probably not worth pursuing here.